### PR TITLE
[Docs] Fix table headers in kyuubi_vs_hive.md

### PR DESCRIPTION
**Pull request by DerickCypho**
### _Why are the changes needed?_

Fix the Markdown formatting for the table that demonstrates the difference between Kyuubi and HiveServer2.
The before table is as shown down below:
image
Document readability improvement.

### CODE REVIEW

**Review by bowenliang123 — APPROVED** [review]
LGTM

**Comment by bowenliang123** [comment]
@DerickCypho Hi, can you fix the style violations by running  `dev/reformat`?

**Comment by DerickCypho** [comment]
> @DerickCypho Hi, can you fix the style violations by running `dev/reformat`?

Sure, appreciate the speedy response.

**Comment by DerickCypho** [comment]
Hi there, I did try run the `/dev/reformat` but didn't work out as expected, it kept reporting as "The requested profile cannot be activated since it doesn't exist" and "no plugin found for the prefix 'spotless' in the current project and in the plugin groups". 

Sorry for the extra effort, I'm new to Kyuubi and hope to utilize it for project later to replace the thrift server.

> @DerickCypho Hi, can you fix the style violations by running `dev/reformat`? And I'm wondering whether the lint tool effects display of the table.

**Comment by pan3793** [comment]
@DerickCypho you should `cd $PROJECT_DIR` before running the `dev/reformat`

**Comment by DerickCypho** [comment]
> @DerickCypho you should `cd $PROJECT_DIR` before running the `dev/reformat`

Thanks, it really helped, I cd-ed into the dev dir, thought it might work, appreciate that. I will try use lint to see if it will have an impact on the final output formatting as well.

File: docs/overview/kyuubi_vs_hive.md [hunk 1]
```diff
@@ -32,16 +32,15 @@ have multiple reducer stages.
 
 ## Differences Between Kyuubi and HiveServer2
```
Comment (yaooqinn) [inline docs/overview/kyuubi_vs_hive.md:34]: I think we should remove this section. It seems a bit offensive

**Comment by yaooqinn** [comment]
thanks, @DerickCypho for your contribution.

File: docs/overview/kyuubi_vs_hive.md (hunk 1 above)
Comment (DerickCypho) [reply docs/overview/kyuubi_vs_hive.md:34]: Actually for me as a newbie to Kyuubi, this section helped me understand the advantages and some major differences in between

**Comment by pan3793** [comment]
So there are issues on markdown formmatter, let's remove it then @bowenliang123 

@DerickCypho please hold on, we need another PR to disable the markdown linter first.

**Comment by bowenliang123** [comment]
If the reformat tool causes display failure,  we might have to remove the markdown rules from the spotless plugin.

File: docs/overview/kyuubi_vs_hive.md [hunk 2]
```diff
@@ -32,16 +32,15 @@ have multiple reducer stages.
...
 
-- 
 
-|             Kyuubi             |                                   HiveServer2                                   |
+|| Kyuubi | HiveServer2 |
```
Comment (bowenliang123) [inline docs/overview/kyuubi_vs_hive.md:36]:
```suggestion
|                              |                                     Kyuubi                                      |                                                                            HiveServer2                                                                             |
```

File: docs/overview/kyuubi_vs_hive.md [hunk 3]
```diff
@@ -32,16 +32,15 @@ have multiple reducer stages.
...
+|**Engine** | up to Spark 3.x                                                                 | MapReduce/[up to Spark 2.3](https://cwiki.apache.org/confluence/display/Hive/Hive+on+Spark%3A+Getting+Started#HiveonSpark:GettingStarted-VersionCompatibility)/Tez |
+| **Performance**| High                                                                            | Low                                                                                                                                                                |
+|**Compatibility with Spark**| Good                                                                            | Bad(need to rebuild on a specific version)                                                                                                                         |
+| **Data Types** | [Spark Data Types](https://spark.apache.org/docs/latest/sql-ref-datatypes.html) | [Hive Data Types](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types)                                                                           |
```
Comment (bowenliang123) [inline docs/overview/kyuubi_vs_hive.md:43]:
```suggestion
| **Language**                 | Spark SQL                                                                       | Hive QL                                                                                                                                                            |
| **Optimizer**                | Spark SQL Catalyst                                                              | Hive Optimizer                                                                                                                                                     |
| **Engine**                   | up to Spark 3.x                                                                 | MapReduce/[up to Spark 2.3](https://cwiki.apache.org/confluence/display/Hive/Hive+on+Spark%3A+Getting+Started#HiveonSpark:GettingStarted-VersionCompatibility)/Tez |
| **Performance**              | High                                                                            | Low                                                                                                                                                                |
| **Compatibility with Spark** | Good                                                                            | Bad(need to rebuild on a specific version)                                                                                                                         |
| **Data Types**               | [Spark Data Types](https://spark.apache.org/docs/latest/sql-ref-datatypes.html) | [Hive Data Types](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types)                                                                           |
```

**Comment by bowenliang123** [comment]
@DerickCypho Could pull the commits again and try to verify the changes I just applied?

After my second checks, the problem doesn't comes with the markdown style rules, but the table content itself, with :
1. not matched column numbers in header
2. the strong style of first column is not conforming to markdown syntax

File: docs/overview/kyuubi_vs_hive.md [hunk 4]
```diff
@@ -32,16 +32,15 @@ have multiple reducer stages.
 
 ## Differences Between Kyuubi and HiveServer2
 
-- 
 
-|             Kyuubi             |                                   HiveServer2                                   |
+|                              |                                     Kyuubi                                      |                                                                            HiveServer2                                                                             |
 |--------------------------------|---------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (bowenliang123) [inline docs/overview/kyuubi_vs_hive.md:37]:
```suggestion
|------------------------------|---------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

File: docs/overview/kyuubi_vs_hive.md [hunk 5]
```diff
@@ -32,16 +32,15 @@ have multiple reducer stages.
...
-| ** Engine **                   | up to Spark 3.x                                                                 | MapReduce/[up to Spark 2.3](https://cwiki.apache.org/confluence/display/Hive/Hive+on+Spark%3A+Getting+Started#HiveonSpark:GettingStarted-VersionCompatibility)/Tez |
-| ** Performance **              | High                                                                            | Low                                                                                                                                                                |
-| ** Compatibility with Spark ** | Good                                                                            | Bad(need to rebuild on a specific version)                                                                                                                         |
-| ** Data Types **               | [Spark Data Types](https://spark.apache.org/docs/latest/sql-ref-datatypes.html) | [Hive Data Types](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types)                                                                           |
+
```
Comment (bowenliang123) [inline docs/overview/kyuubi_vs_hive.md:35]:
```suggestion
```

**Comment by bowenliang123** [comment]
@DerickCypho Hi, Derick. Could you try to pull the latest commits and validated the HTML output again? It should work fine now.

It turns out the violation in markdown contents themselves , but not the markdown rules.
1. incorrect column header numbers
2. incorrect usage of bold style text in table.

**Comment by DerickCypho** [comment]
> thanks, @DerickCypho for your contribution.

> @DerickCypho Hi, Derick. Could you try to pull the latest commits and validated the HTML output again? It should work fine now.
> 
> It turns out the violation in markdown contents themselves , but not the markdown rules.
> 
> 1. incorrect column header numbers
> 2. incorrect usage of bold style text in table.

Yes I found the same result after I've tried re-patched them after all those changes via the reformat script or other re-formatters on hand. 
Appreciate the follow-up and it worked like a charm on my side too. Thanks a lot and I'll have my go later on the project for thrift server replacement on OLAP stuff.

Sincerely,
Derick

**Review by bowenliang123 — APPROVED** [review]
LGTM. Thanks for the effort and double-checks. Let's do the merging this PR to see whether it works as expected.

**Comment by pan3793** [comment]
Thanks all, merged to master/1.7

**Comment by DerickCypho** [comment]
> Thanks all, merged to master/1.7

My appreciation.

**Pull request closed without merging**